docker/util/run.bzl

```diff
@@ -49,74 +50,50 @@ def _extract_impl(
                     by another action.
     """
     name = name or ctx.label.name
-    image = image or ctx.file.image
-    commands = commands or ctx.attr.commands
-    docker_run_flags = docker_run_flags or ctx.attr.docker_run_flags
     extract_file = extract_file or ctx.attr.extract_file
     output_file = output_file or ctx.outputs.out
-    script = script_file or ctx.outputs.script
-
-    toolchain_info = ctx.toolchains["@io_bazel_rules_docker//toolchains/docker:toolchain_type"].info
+    script_file = script_file or ctx.outputs.script
+    output_tarball = ctx.actions.declare_file(name + ".tar")
+    output_digest = ctx.actions.declare_file(name + ".digest")
+    output_config = ctx.actions.declare_file(name + ".config")
+    output_layer = ctx.actions.declare_file(name + ".layer")
 
-    # Generate a shell script to execute the run statement
-    ctx.actions.expand_template(
-        template = ctx.file._extract_tpl,
-        output = script,
-        substitutions = {
-            "%{commands}": _process_commands(commands),
-            "%{docker_flags}": " ".join(toolchain_info.docker_flags),
-            "%{docker_run_flags}": " ".join(docker_run_flags),
-            "%{docker_tool_path}": toolchain_info.tool_path,
-            "%{extract_file}": extract_file,
-            "%{image_id_extractor_path}": ctx.executable._extract_image_id.path,
-            "%{image_tar}": image.path,
-            "%{output}": output_file.path,
-        },
-        is_executable = True,
+    image_result = _image.implementation(
+        ctx,
+        name,
+        base = base,
+        cmd = cmd,
```

Comment:
The benefit here is that this allows the build to define an entry point other than `sh -c`.  For example, it allows to simulate a check such as:

`docker run amazon/aws-cli --version`

(The key with this example is that `amazon/aws-cli` has an entry point of `aws` rather than `sh`, so this is running a command `aws --version` )

Reply:
This is now backwards compatible so you can keep using `commands`.